addons/sale/models/sale_order.py

```diff
@@ -1412,7 +1412,7 @@ def action_view_invoice(self, invoices=False):
         return action
 
     def _get_invoice_grouping_keys(self):
-        return ['company_id', 'partner_id', 'currency_id']
+        return ['company_id', 'partner_id', 'currency_id', 'partner_shipping_id']
```

Comment:
You should adept this comment also
 `2) Manage 'grouped' parameter: group by (partner_id, currency_id).`

And is sequencing have any effect like if you write `partner_shipping_id`  before `currency_id`? :thinking:

Reply:
Yes, the sequence will affect the grouping. Therefore, I am placing the `partner_shipping_id` before the `currency_id`. Thank you for pointing this out!